Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -19,10 +22,7 @@ internal static class WebHeaderCollectionExtensions
         /// <returns>Value against the given parameters.</returns>
         public static string GetNameValueHeaderValue(this NameValueCollection headers, string headerName, string keyName)
         {
-            if (string.IsNullOrEmpty(headerName))
-            {
-                throw new ArgumentNullException(nameof(headerName));
-            }
+            Debug.Assert(headerName != null, "headerName must not be null");
```

Comment:
If a debug build assert is all that is left, what is the point of even passing headerName as a parameter at all?  It wasn't used in the original call either other than to throw if it was null or empty which seems pointless.